physionet-django/events/views.py

```diff
@@ -238,3 +238,54 @@ def event_detail(request, event_slug):
          'is_waitlisted': is_waitlisted,
          'event_datasets': event_datasets,
          })
+
+
+@login_required
+def manage_co_hosts(request):
+    """
+    Manage co-hosts of an event
+    """
+    user = request.user
+
+    if request.method == 'POST' and request.META.get('HTTP_X_REQUESTED_WITH') == 'XMLHttpRequest':
+        participant_id = request.POST.get('participant_id')
+
+        event_slug = request.POST.get('event_slug')
+        event = get_object_or_404(Event, slug=event_slug)
+
+        if not event.host == user:
+            return JsonResponse({'error': 'You are not the host of this event'}, status=403)
+
+        if event.end_date < datetime.now().date():
+            return JsonResponse({'error': 'You cannot manage co-hosts of an event that has ended'}, status=403)
+
+        if not event.participants.filter(id=participant_id).exists():
+            return JsonResponse({'error': 'User is not a participant of this event'}, status=403)
+
+        participant = event.participants.get(id=participant_id)
+
+        if 'Remove cohost' in request.POST.get('submit'):
+            if not participant.is_cohost:
+                return JsonResponse({'error': 'User is not a cohost of this event'}, status=403)
+            participant.is_cohost = False
+            participant.save()
+            notification.notify_event_cohost_cohost_status_change(request=request, cohost=participant.user,
+                                                                  event=event, status='Remove cohost')
+            notification.notify_event_host_cohost_status_change(request=request, cohost=participant.user, event=event,
+                                                                status='Remove cohost')
```

Comment:
Passing the status here as a raw string looks very loose - we need some enum value to decide which action we are going to take. Right now if you make a typo anywhere in the codebase, you introduce a bug.

Reply:
Agreed, I've changed it to the actual participant cohost_status.